Assets/Atlas games/Scripts/DoubleXPItem/DoubleXPManager.cs

```diff
@@ -0,0 +1,150 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using UnityEngine;
+
+public class DoubleXPManager : MonoBehaviour
+{
+    static TimeAndDateResponseModel date;
+    static string extractedDate;
+    static string extractedTime;
+    static string dateTimeString;
+
+    public enum DoubleXpDuration
+    {
+        Day,
+        Hour
+    }
+
+    TimeAndDateResponseModel _globalDate;
+    static TimeSpan globalTimeDifference;
+
+    async void Awake()
+    {
+        DontDestroyOnLoad(gameObject);
```

Comment:
برای اطمینان از اینکه فقط یکی از این آبجکت داشته باشیم.
```suggestion
    public static DoubleXPManager self;
    async void Awake()
    {
        if (self == null){
            self = this;
            DontDestroyOnLoad(this);
        } else {
            Destroy(this);
        }
```